src/cli.rs

```diff
@@ -0,0 +1,103 @@
+        .subcommand(
+            Command::new("link")
+                .about("Link microfrontends and run altogether")
+                .args(
+                    vec![
+                        Arg::with_name(options::long::MFE)
+                            .short(options::short::MFE)
+                            .long(options::long::MFE)
+                            .help("Which microfrontends to fetch")
+                            .takes_value(true)
+                            .required(true)
+                            .multiple_values(true)
+                            .value_parser(value_parser!(String)),
+                        Arg::with_name(options::long::TARGET_HOST)
+                            .short(options::short::TARGET_HOST)
+                            .long(options::long::TARGET_HOST)
+                            .help("Where repo is fetched from")
+                            .takes_value(true)
+                            .required(true)
+                            .value_parser(value_parser!(String)),
+                        Arg::with_name(options::long::FORCE_UPDATE_ALL)
+                            .short(options::short::FORCE_UPDATE_ALL)
+                            .long(options::long::FORCE_UPDATE_ALL)
+                            .help("Re-install dependencies for all MFEs")
+                            .takes_value(false)
+                            .required(false),
+                        Arg::with_name(options::long::PULL)
+                            .short(options::short::PULL)
+                            .long(options::long::PULL)
+                            .help("Pull latest changes from desired versions")
+                            .takes_value(false)
+                            .required(false),
+                        //  TODO: arg!(--"force-update" <FORCE UPDATE DESIRED MFE>).required(false),
+                    ],
+                ),
+        )
+}
+
+fn parse_link_command(arg_matches: &ArgMatches) -> LinkCliResult {
+    let target_host = arg_matches.get_one::<String>(options::long::TARGET_HOST).unwrap().to_string();
+    let microfrontend: Vec<_> = arg_matches.values_of(options::long::MFE).unwrap()
+        .map(|str| (*str).to_owned())
+        .collect();
+    let force_update_all = arg_matches.is_present(options::long::FORCE_UPDATE_ALL);
+    let pull = arg_matches.is_present(options::long::PULL);
+
+    LinkCliResult { target_host, microfrontend, force_update_all, pull }
+}
+
+pub fn parse() -> LinkCliResult {
+    let matches = self::initiate().get_matches();
+    let arg_matches = matches.subcommand_matches("link").unwrap();
+
+    self::parse_link_command(&arg_matches)
+}
+
+pub fn link_options_adapter(link_result: LinkCliResult) -> (Microfrontends, Configuration) {
+    let configuration = Configuration {
+        target_host: link_result.target_host,
+        force_update_all: link_result.force_update_all,
+        pull: link_result.pull
+    };
```

Comment:
One other approach could be the pattern syntax https://doc.rust-lang.org/book/ch18-01-all-the-places-for-patterns.html#function-parameters

```suggestion
pub fn link_options_adapter(LinkCliResult { target_host,  force_update_all, pull }: LinkCliResult) -> (Microfrontends, Configuration) {
    let configuration = Configuration {
        target_host,
        force_update_all,
        pull,
    };
```